### DISCUSSION TITLE
Unmet peer rollup@^2.63.0: found 3.2.3

### DISCUSSION THREAD

**Discussion by taco-chen** [directus/directus#16406] [Archive]
### Describe the Bug

After upgrading Directus to the latest version 9.20.4

```
project
└─┬ directus 9.20.4
  └─┬ @directus/extensions-sdk 9.20.4
    ├─┬ rollup-plugin-styles 4.0.0
    │ └── ✕ unmet peer rollup@^2.63.0: found 3.2.3 in @directus/extensions-sdk
    └─┬ rollup-plugin-terser 7.0.2
      └── ✕ unmet peer rollup@^2.0.0: found 3.2.3 in @directus/extensions-sdk
```

### To Reproduce

Update directus version to 9.20.4 in the packages.json file and

```
$ pnpm i
```

**Comment by azrikahar** [comment]
Moving this to discussion as I believe it doesn't affect Directus runtime itself 👍

This should be due to the update to `rollup` from `^2.67.3` to `3.2.3` in https://github.com/directus/directus/pull/16052/files#diff-bf5b5d2513f9e215c8872acc7ae225035fc81ec09a5cee9ddcd1b8708c52a4f1 which now doesn't correctly match the peer dependency version in the affected packages. @nickrum any thoughts on this?

**Comment by eugenio-campos** [comment]
The "rollup-plugin-terser" package is no longer maintained. It could be replaced by the official plugin "@rollup/plugin-terser" which supports rollup v3.

packages/extensions-sdk/package.json
```
- "rollup-plugin-terser": "7.0.2"
+ "@rollup/plugin-terser": "0.1.0"
```

packages/extensions-sdk/src/cli/commands/build.ts
```
- import { terser } from 'rollup-plugin-terser';
+ import terser from '@rollup/plugin-terser';
```

**Reply by rijkvanzanten** [reply]
Thanks for the suggestion! Now the wait is just for https://github.com/Anidetrix/rollup-plugin-styles/issues/224

**Reply by adanielyan** [reply]
How do we update directus in the meantime? Running `pnpm update` shows this warning and the directus doesn't get updated to 9.22.4:
```
WARN  Issues with peer dependencies found
.
└─┬ directus 9.22.4
  └─┬ @directus/extensions-sdk 9.22.4
    └─┬ rollup-plugin-styles 4.0.0
      └── ✕ unmet peer rollup@^2.63.0: found 3.7.5 in @directus/extensions-sdk
```

**Reply by paescuj** [reply]
> How do we update directus in the meantime? Running `pnpm update` shows this warning and the directus doesn't get updated to 9.22.4:
> 
> ```
> WARN  Issues with peer dependencies found
> .
> └─┬ directus 9.22.4
>   └─┬ @directus/extensions-sdk 9.22.4
>     └─┬ rollup-plugin-styles 4.0.0
>       └── ✕ unmet peer rollup@^2.63.0: found 3.7.5 in @directus/extensions-sdk
> ```

This is just a warning and shouldn't prevent the update itself.

**Reply by adanielyan** [reply]
That's what I would expect. However I can't get past through the v9.22.3.

**Reply by CallMeLaNN** [reply]
> That's what I would expect. However I can't get past through the v9.22.3.

You can go trough the upgrade doc https://docs.directus.io/self-hosted/upgrades-migrations.html
and make sure migrate the db also as mentioned in there and also check the release or any breaking change. You probably need some help on the cli. It's totally doesn't related to this warning.

**Reply by paescuj** [reply]
@adanielyan Are you checking the version via the app? Because version v9.22.4 is wrongly displayed as 9.22.3 - see https://github.com/directus/directus/issues/17115

**Reply by adanielyan** [reply]
Ah, I didn't know about this issue! Thank you @paescuj for pointing it out!

**Comment by taco-chen** [comment]
It seems the issue has not bee solved yet in the directus v10.8.2

```
 apps/cms
 └─┬ *** 10.8.2
   └─┬ @***/api 14.1.2
     └─┬ @***/extensions-sdk 10.2.0
       └─┬ rollup-plugin-styles 4.0.0
         └── ✕ unmet peer rollup@^2.63.0: found 3.22.0 in @***/extensions-sdk
```

**Reply by paescuj** [reply]
This warning can be ignored 👍

**Comment by rokdd** [comment]
> This warning can be ignored 👍

What does it mean to ignore? I can not update or rebuild directus and currentkly no glue how to solve it.
![image](https://github.com/directus/directus/assets/640866/37266c3b-9435-4ae0-b64b-25f0d5831bb0)

**Reply by paescuj** [reply]
This is just a warning, not an actual error.
If you have trouble updating/rebuilding Directus it must be related to another issue.
Please open an issue report with all the necessary information (the exact steps to reproduce), so that we can take a closer look.

**Comment by Joonel** [comment]
I can't build (or dev) an extension with `directus-extension build` because of this when using `pnpm i`. `npm i`seems to work.
Directus app and the extension are in the same repo (extension is in `./extensions/extension-name` as designed). 
```
> directus-extension build

node:internal/modules/run_main:129
    triggerUncaughtException(
    ^

Error: Cannot find package 'C:\Users\user\source\repos\directus\extensions\generate-slug\node_modules\.pnpm\@directus+extensions-sdk@11.0.9_@types+node@20.14.11_@unhead+vue@1.9.16_vue@3.4.27__knex@3.1._qzdeh7wfjdh5ua3fvo6ocwr6aq\node_modules\rollup-plugin-styles\package.json'
imported from C:\Users\user\source\repos\directus\extensions\generate-slug\node_modules\.pnpm\@directus+extensions-sdk@11.0.9_@types+node@20.14.11_@unhead+vue@1.9.16_vue@3.4.27__knex@3.1._qzdeh7wfjdh5ua3fvo6ocwr6aq\node_modules\@directus\extensions-sdk\dist\cli\commands\build.js
Did you mean to import "rollup-plugin-styles/dist/index.js"?
    at legacyMainResolve (node:internal/modules/esm/resolve:215:26)
    at packageResolve (node:internal/modules/esm/resolve:841:14)
    at moduleResolve (node:internal/modules/esm/resolve:927:18)
    at defaultResolve (node:internal/modules/esm/resolve:1157:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:383:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:352:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:227:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)
    at link (node:internal/modules/esm/module_job:86:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

**Comment by JoshuaCWebDeveloper** [comment]
This should be converted back into an issue, as it is caused by the code in this repo, not anyone else's.

It isn't great practice to ask users to simply ignore the warning. It is a warning for a good reason and inconveniences devs trying to maintain clean projects. This warning is caused by the `@directus/extensions-sdk` package including both `rollup@3.29.4` and `rollup-plugin-styles@4.0.0`, which conflict with each other (`rollup-plugin-styles` doesn't currently officially support `rollup@3`). https://github.com/directus/directus/blob/2e893f9c576d5a02506272fe2c0bcc12e6c58768/packages/extensions-sdk/package.json#L54-L56

This warning can be fixed by either:
1) Replacing `rollup-plugin-styles` with a better maintained alternative, or
2) Updating `@directus/extensions-sdk` to override the rollup version for `rollup-plugin-styles` by adding an override to the `package.json` like so:
```json
"overrides": {
  "rollup-plugin-styles": {
    "rollup": "3.29.4"
  }
}
```
https://github.com/Anidetrix/rollup-plugin-styles/issues/224 can also be monitored for updates.

As a workaround, users can apply the above override to their own `package.json` files to silence the warning coming from @directus/extensions-sdk`.
